File: src/leetcode/2026/06/Solution999.java
public class Solution999 {
    public int numRookCaptures(char[][] board) {
        int count = 0;
        int rookRow = -1, rookCol = -1;

        // Find the position of the rook
        for (int i = 0; i < 8; i++) {
            for (int j = 0; j < 8; j++) {
                if (board[i][j] == 'R') {
                    rookRow = i;
                    rookCol = j;
                    break;
                }
            }
            if (rookRow != -1) {
                break;
            }
        }

        // Check upwards
        for (int i = rookRow - 1; i >= 0; i--) {
            if (board[i][rookCol] == 'B') {
                break; // Blocked by a bishop
            }
            if (board[i][rookCol] == 'p') {
                count++;
                break; // Captured a pawn
            }
        }

        // Check downwards
        for (int i = rookRow + 1; i < 8; i++) {
            if (board[i][rookCol] == 'B') {
                break; // Blocked by a bishop
            }
            if (board[i][rookCol] == 'p') {
                count++;
                break; // Captured a pawn
            }
        }

        // Check leftwards
        for (int j = rookCol - 1; j >= 0; j--) {
            if (board[rookRow][j] == 'B') {
                break; // Blocked by a bishop
            }
            if (board[rookRow][j] == 'p') {
                count++;
                break; // Captured a pawn
            }
        }

        // Check rightwards
        for (int j = rookCol + 1; j < 8; j++) {
            if (board[rookRow][j] == 'B') {
                break; // Blocked by a bishop
            }
            if (board[rookRow][j] == 'p') {
                count++;
                break; // Captured a pawn
            }
        }

        return count;
    }

    public static void main(String[] args) {
        Solution999 solution = new Solution999();

        char[][] board1 = {
                { '.', '.', '.', '.', '.', '.', '.', '.' },
                { '.', '.', '.', 'p', '.', '.', '.', '.' },
                { '.', '.', '.', 'R', '.', '.', '.', 'p' },
                { '.', '.', '.', '.', '.', '.', '.', '.' },
                { '.', '.', '.', 'p', '.', '.', '.', '.' },
                { '.', '.', '.', '.', '.', '.', '.', '.' },
                { '.', '.', '.', 'p', '.', '.', '.', '.' },
                { '.', '.', '.', 'B', '.', '.', '.', '.' }
        };
        System.out.println("Output for test case 1: " + solution.numRookCaptures(board1)); // Output: 3

        char[][] board2 = {
                { '.', '.', '.', '.', '.', '.', '.', '.' },
                { '.', '.', '.', 'p', '.', '.', '.', '.' },
                { '.', '.', '.', 'p', '.', '.', '.', '.' },
                { 'p', 'p', '.', 'R', '.', 'p', 'B', '.' },
                { '.', '.', '.', '.', '.', '.', '.', '.' },
                { '.', '.', '.', 'B', '.', '.', '.', '.' },
                { '.', '.', '.', 'p', '.', '.', '.', '.' },
                { '.', '.', '.', '.', '.', '.', '.', '.' }
        };
        System.out.println("Output for test case 2: " + solution.numRookCaptures(board2)); // Output: 3

        char[][] board3 = {
                { '.', '.', '.', '.', '.', '.', '.', '.' },
                { '.', 'p', 'p', 'p', 'p', 'p', '.', '.' },
                { '.', 'p', 'B', 'B', 'B', 'p', '.', '.' },
                { '.', 'p', 'B', 'R', 'B', 'p', '.', '.' },
                { '.', 'p', 'B', 'B', 'B', 'p', '.', '.' },
                { '.', 'p', 'p', 'p', 'p', 'p', '.', '.' },
                { '.', '.', '.', '.', '.', '.', '.', '.' },
                { '.', '.', '.', '.', '.', '.', '.', '.' }
        };
        System.out.println("Output for test case 3: " + solution.numRookCaptures(board3)); // Output: 0
    }
}

File: src/leetcode/2026/06/Solution999.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution999 {
public:
    int numRookCaptures(vector<vector<char>>& board) {
        int count = 0;
        int rookX = -1, rookY = -1;
        for (int i = 0; i < 8; i++) {
            for (int j = 0; j < 8; j++) {
                if (board[i][j] == 'R') {
                    rookX = i;
                    rookY = j;
                    break;
                }
            }
            if (rookX != -1) break;
        }

        // Check upwards
        for (int i = rookX - 1; i >= 0; i--) {
            if (board[i][rookY] == 'B') break;
            if (board[i][rookY] == 'p') {
                count++;
                break;
            }
        }

        // Check downwards
        for (int i = rookX + 1; i < 8; i++) {
            if (board[i][rookY] == 'B') break;
            if (board[i][rookY] == 'p') {
                count++;
                break;
            }
        }

        // Check left
        for (int j = rookY - 1; j >= 0; j--) {
            if (board[rookX][j] == 'B') break;
            if (board[rookX][j] == 'p') {
                count++;
                break;
            }
        }

        // Check right
        for (int j = rookY + 1; j < 8; j++) {
            if (board[rookX][j] == 'B') break;
            if (board[rookX][j] == 'p') {
                count++;
                break;
            }
        }

        return count;
    }
};

int main() {
    Solution999 solution;

    vector<vector<char>> board = {
        {'.', '.', '.', '.', '.', '.', '.', '.'},
        {'.', '.', '.', 'p', '.', '.', '.', '.'},
        {'.', '.', '.', 'R', '.', '.', '.', 'p'},
        {'.', '.', '.', '.', '.', '.', '.', '.'},
        {'.', '.', '.', '.', '.', '.', '.', '.'},
        {'.', '.', '.', 'p', '.', '.', '.', '.'},
        {'.', '.', '.', 'B', '.', '.', '.', '.'},
        {'.', '.', '.', 'p', '.', '.', '.', '.'}
    };

    int captures = solution.numRookCaptures(board);
    cout << "Number of pawns the rook can capture: " << captures << endl;

    return 0;
}